simplq/src/components/pages/Admin/TokenList.jsx

```diff
@@ -1,40 +1,23 @@
 import React from 'react';
-import LoadingIndicator from 'components/common/LoadingIndicator';
+import Loading from 'components/common/Loading/Loading';
 import Token from './Token';
 import styles from './admin.module.scss';
 
-function TokenList(props) {
-  const queueId = props.queueId;
-  let listContent = null;
-
-  if (!props.tokens) {
-    listContent = (
-      <div className={styles['token-list']}>
-        <LoadingIndicator />
-      </div>
+function TokenList({ tokens }) {
+  const Dispaly = () =>
+    tokens.length === 0 ? (
+      <p>Your queue has been created and is currently empty. Waiting for people to join...</p>
+    ) : (
+      tokens.map((token) => <Token token={token} key={token.tokenId} />)
     );
-  } else if (props.tokens.length === 0) {
-    listContent = (
-      <div className={styles['token-list']}>
-        <p>Your queue has been created and is currently empty. Waiting for people to join...</p>
-      </div>
-    );
-  } else {
-    listContent = (
+
+  return (
+    <Loading isLoading={tokens === undefined}>
       <div className={styles['token-list']}>
-        {props.tokens.map((token) => (
-          <Token
-            removeTokenHandler={props.removeTokenHandler}
-            token={token}
-            queueId={queueId}
-            key={token.tokenId}
-          />
-        ))}
+        <Dispaly />
```

Comment:
Typo. Maybe call it Cards or something else?

Reply:
I'm open to suggestions. `ToListOrNotToList` 🤣

Reply:
`ListContent`?

Reply:
Yeah, that works. I'm thinking if we should move all the token logic into this file. That way, if a token is added or removed, only the token list rerenders instead of the whole admin page. Do you see any problems with that?

Reply:
Both header and token list depend on `getSelectedQueue`. Since only the redux store is changing, and only the tokens field is changing, I believe only the token list will rerender.

Reply:
Okay, got it

Reply:
I'm wrong, Admin page renders every time.

Reply:
Yes, I just checked with some logs, seems like the admin component does rerender everytime. I'll make this change soon.